app/views/shared/_flashes.html.erb

```diff
@@ -1,13 +1,12 @@
-<%
-flash.
+<% flash.
   to_hash.
   slice(*ApplicationController::FLASH_KEYS + ApplicationController::FLASH_KEY_MAP.keys).
   each do |type, message| %>
-  <% if message.present? %>
-    <%= render 'shared/alert', {
-      type: ApplicationController::FLASH_KEY_MAP.fetch(type, type),
-      message: message.html_safe,
-      class: 'margin-bottom-4',
-    } %>
-  <% end %>
+    <% if message.present? %>
+      <%= render AlertComponent.new(
+            type: ApplicationController::FLASH_KEY_MAP.fetch(type, type),
```

Comment:
what if we moved this constant into the component or made a method out of it? ApplicationController seems like an interesting place for this kind of display mapping?

Some ideas:

```
AlertComponent::FLASH_TYPE_TO_ALERT_TYPE
```

```
AlertComponent.type(flash: type)
```

Reply:
> what if we moved this constant into the component or made a method out of it? ApplicationController seems like an interesting place for this kind of display mapping?
> 
> Some ideas:
> 
> ```
> AlertComponent::FLASH_TYPE_TO_ALERT_TYPE
> ```
> 
> ```
> AlertComponent.type(flash: type)
> ```

Yeah, one possible downside is that it ties the component to Rails, where this might be a perfect example of a component to eventually pull out and make available in the design system for static sites, etc. I do agree with your general point though, and maybe it's fine not to shy completely away from Rails-isms in the component.

Reply:
Hm, another option could be to adapt `app/views/shared/_flashes.html.erb` into a "Flashes" component which composes Alert and absorbs this mapping logic.

Reply:
That works too! The thing that was jumping out the most to me was the ApplicationController doing this much, because that controller never really renders anything

Reply:
> That works too! The thing that was jumping out the most to me was the ApplicationController doing this much, because that controller never really renders anything

Yep, as the person who wrote that code, I totally agree 😄

Reply:
Added FlashComponent in fc17575.